src/commands/service/create.ts

```diff
@@ -41,26 +42,12 @@ export const createCommandDecorator = (command: commander.Command): void => {
       const projectPath = process.cwd();
       try {
         // Type check all parsed command line args here.
-        if (typeof serviceName !== "string") {
-          throw new Error(
-            `serviceName must be of type 'string', ${typeof serviceName} given.`
-          );
-        }
-        if (typeof packagesDir !== "string") {
-          throw new Error(
-            `packagesDir must be of type 'string', ${typeof packagesDir} given.`
-          );
-        }
-        if (typeof maintainerName !== "string") {
-          throw new Error(
-            `maintainerName must be of type 'string', ${typeof maintainerName} given.`
-          );
-        }
-        if (typeof maintainerEmail !== "string") {
-          throw new Error(
-            `maintainerEmail must be of type 'string', ${typeof maintainerEmail} given.`
-          );
-        }
+        const stringType = "string";
+        guardTypeOf(serviceName, stringType);
+        guardTypeOf(packagesDir, stringType);
+        guardTypeOf(maintainerName, stringType);
+        guardTypeOf(maintainerEmail, stringType);
```

Comment:
Does `guardTypeOf` actually act as a type guard for any of the variables?
Following lines 49, does the compiler actually know `serviceName`, `packagesDir`, etc... are of type `string`?
I don't believe the current setup will actually allow the compiler to know the types of the variables (ie; their still `any`) after the `guardTypeOf` calls; whereas the compiler knows their of type `string` using the `if (typeof maintainerEmail !== "string") {...` syntax.

Reply:
Just played with this a bit, just as I feared

Fails:

```ts
const log = (msg: string) => console.log(msg);
const foo: unknown = "i'm a string";
guardTypeOf(foo, "string");
log(foo);
```

Passes:

```ts
const log = (msg: string) => console.log(msg);
const foo: unknown = "i'm a string";
if (typeof foo !== "string") {
  throw new Error("Expected string");
}
log(foo);
```

We should maintain to just use vanilla TS typeguards for now (`if (typeof ...`)).

Reply:
I'm ok with using vanilla guards.  But is there a way to abstract it out to keep the main logic simple?

Reply:
Sadly not yet. TS is planning on supporting `assert`ion guards which do exactly this in TS 3.7 https://devblogs.microsoft.com/typescript/announcing-typescript-3-7-beta/

But until then, the compiler isn't smart enough to really support a clean abstraction :(